content/docs/user-guide/how-to/add-deps-or-outs-to-a-stage.md

```diff
@@ -0,0 +1,55 @@
+# How to Add Dependencies or Outputs
+
+We have executed a stage, but later notice that some of the files/directories it
+uses as dependencies or creates as outputs are missing from `dvc.yaml`...
+
+To add files or directories as <abbr>dependencies</abbr> or <abbr>outputs</abbr>
+to an existing stage without re-executing it (which can be expensive and is
+unnecessary) you can either edit the `dvc.yaml` file directly, or use `dvc run`
+with the `-f` and `--no-exec` options to the same end.
+
+After the update, use `dvc commit` to save any output files that already exist
```

Comment:
This should go as a very specific note if it's a general case. In general case you edit, and run `dvc repro`. `dvc commit` should be considered as a hack in most cases. Users should understand what are they doing when they use it (potentially break reproducibility).

Reply:
OK, I started with a conditional now, but I don't imagine this is such a rare edge case. May even be the norm that you realize the stage needs an update only after haven forgotten to define all the outs and executing it that way.

I don't know if the generic case of just continuing to edit dvc.yaml before having reproed the pipeline even needs a doc. It will be pretty obvious for those who create stages by hand by editing dvc.yaml, no? People who may find this useful will have already done a repro or used run from the beginning, I assume.